sdk/benches/sort.rs

```diff
@@ -0,0 +1,72 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion, Throughput};
+use std::time::Duration;
+use surrealdb::dbs::Session;
+use surrealdb::kvs::Datastore;
+use surrealdb_core::sql::Value;
+use tokio::runtime::Runtime;
+
+fn bench_sort(c: &mut Criterion) {
+	let rt = Runtime::new().unwrap();
+
+	let mut group = c.benchmark_group("sort");
+	group.throughput(Throughput::Elements(1));
+	group.sample_size(10);
+	group.measurement_time(Duration::from_secs(15));
+
+	let i = rt.block_on(prepare_data(9999));
```

Comment:
Maybe we increase this to more than 10000 to know better results?

Reply:
Parallel sorting becomes beneficial only when the overhead of dividing the work across multiple threads is outweighed by the time saved in sorting. This bench just checked that 10000 is enough to start having some benefit

Reply:
Ack, but my question was more about how much time does this implementation actually save us with large datasets, because with the benchmark (even though it is about checking at what length of records using rayon make more sense) it isn't so clear. I ran the benchmark locally and it gave me this result

```
sort/sort               time:   [57.999 ms 58.436 ms 59.166 ms]
                        thrpt:  [16.902  elem/s 17.113  elem/s 17.242  elem/s]

sort/sort-parallel      time:   [55.657 ms 56.521 ms 57.547 ms]
                        thrpt:  [17.377  elem/s 17.693  elem/s 17.967  elem/s]

sort/sort-rayon         time:   [55.924 ms 56.595 ms 57.246 ms]
                        thrpt:  [17.469  elem/s 17.669  elem/s 17.881  elem/s]

sort/sort-rayon-parallel
                        time:   [53.087 ms 60.356 ms 69.705 ms]
                        thrpt:  [14.346  elem/s 16.568  elem/s 18.837  elem/s]
```

Reply:
I'll run it with higher number of records and check